jvm/src/main/scala/org/apache/spark/sql/execution/ColumnarCollapseCodegenStages.scala

```diff
@@ -121,13 +122,19 @@ case class ColumnarCollapseCodegenStages(
     plan match {
       case p if !supportTransform(p) =>
         new ColumnarInputAdapter(insertWholeStageTransformer(p))
+      // The children of BaseJoinExec should do transformer separately.
+      case j if j.isInstanceOf[BaseJoinExec] =>
+        j.withNewChildren(j.children.map(c => insertWholeStageTransformer(c)))
       case p =>
         p.withNewChildren(p.children.map(insertInputAdapter))
     }
   }
 
   private def insertWholeStageTransformer(plan: SparkPlan): SparkPlan = {
     plan match {
+      // The children of BaseJoinExec should do transformer separately.
+      case t: TransformSupport if t.isInstanceOf[BaseJoinExec] =>
+        t.withNewChildren(t.children.map(insertWholeStageTransformer))
```

Comment:
This seems not right and should be removed. We need to insert WholeStageTransformerExec for Join.

Reply:
Fixed